src/bootstrap/config.rs

```diff
@@ -678,7 +679,7 @@ impl<T> Merge for Option<T> {
                             panic!("overriding existing option")
                         } else {
                             eprintln!("overriding existing option");
-                            crate::detail_exit(2);
+                            detail_exit(2, cfg!(test));
```

Comment:
Could `detail_exit` be a macro or somehow otherwise take an exit message? Then we could avoid duplicating the message in `panic`/`eprintln + exit` and handle it internally

Reply:
We already have this function for that
https://github.com/rust-lang/rust/blob/c6e42f5a38c7997da5cbb966aa93c3335c1805a8/src/tools/build_helper/src/util.rs#L15-L18

Which will make this one useless by writing to stderr twice